integrations/operator/controllers/resources/role_controller.go

```diff
@@ -123,7 +125,10 @@ func (r *RoleReconciler) Upsert(ctx context.Context, obj kclient.Object) error {
 
 	// Converting the Kubernetes resource into a Teleport one, checking potential ownership issues.
 	teleportResource := k8sResource.ToTeleport()
-	teleportClient, err := r.TeleportClientAccessor(ctx)
+	teleportClient, release, err := r.TeleportClientAccessor(ctx)
+	if err == nil {
+		defer release()
+	}
```

Comment:
Why are we not returning an error when `err != nil` ?

Reply:
Because we are processing the error and reporting it in the status, so we don't want to exit yet. A few lines below the error is aggregated with the status update error. `err` handling continues below.

We have 4 cases:
- err != nil, updaterErr != nil
- err != nil, updaterErr == nil
- err == nil, updateErr != nil
- err == nil, updaterErr != nil

We must release the teleportClient only on the last two cases, we can do two `release` invocation (one in the `if err != nil || updateErr != nil {` block, and one after) or do a single one before, then continue to process errors.

Reply:
I was scrolling and the pattern was different for the Role
But it seems it was already the case anyway
:+1: